client/src/modules/Score/components/ScoreTable/index.tsx

```diff
@@ -84,17 +88,19 @@ export function ScoreTable(props: Props) {
     cityName,
     mentor,
     handleSettings: () => setIsVisibleSettings(true),
-    taskColumns: getTaskColumns(courseTasks).filter(column => !notVisibleColumns?.includes(column.name)),
+    taskColumns: getTaskColumns(courseTasks),
   });
 
+  const getVisibleColumns = (columns: any[]) => columns.filter(column => !notVisibleColumns?.includes(column.name));
+
   const [notVisibleColumns, setNotVisibleColumns] = useLocalStorage<string[]>('notVisibleColumns', []);
 
   const handleModalCancel = () => {
     setIsVisibleSettings(!isVisibleSetting);
   };
 
   const handleModalOk = (values: Store) => {
-    setNotVisibleColumns(Object.keys(values).filter((value: string) => values[value] === false));
+    setNotVisibleColumns(Object.keys(values).filter((value: string) => !values[value]));
```

Comment:
The `!values[value]` operation takes longer than `values [value] === false`. What is the point of replacing it?

Reply:
done